kroxylicious/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -146,28 +115,59 @@ else if (requestFilterResult.message() != null) {
         }
     }
 
+    private CompletableFuture<RequestFilterResult> executeWrite(ChannelHandlerContext ctx, DecodedRequestFrame<?> decodedFrame,
+                                                                DefaultFilterContext filterContext,
+                                                                CompletableFuture<RequestFilterResult> withTimeout) {
+        return withTimeout.whenComplete((requestFilterResult, t) -> {
```

Comment:
nit:
```suggestion
    private CompletableFuture<RequestFilterResult> executeWrite(ChannelHandlerContext ctx, DecodedRequestFrame<?> decodedFrame,
                                                                DefaultFilterContext filterContext,
                                                                CompletableFuture<RequestFilterResult> pendingStage) {
        return pendingStage.whenComplete((requestFilterResult, t) -> {
```

Reply:
+1 to a rename.   From the point of view of this method, it doesn't know if the stage is deferred or not, so I think just plain `stage` would be sufficient.

Reply:
I included it in https://github.com/robobario/kproxy/pull/8